superset-frontend/src/components/Chart/Chart.jsx

```diff
@@ -232,16 +246,50 @@ class Chart extends React.PureComponent {
     );
   }
 
+  renderSpinner(databaseName) {
+    const message =
```

Comment:
Maybe `const message = databaseName ? t('Waiting on database...') : t('Waiting on %s', databaseName);` just to be safe as I don't know if the database name can be `null` or empty.